src/views/RecIM/components/Forms/Form/index.jsx

```diff
@@ -0,0 +1,214 @@
+
+  const handleSetError = (field, condition) => {
+    const getCurrentErrorStatus = (currentValue) => {
+      return {
+        ...currentValue,
+        [field]: condition,
+      };
+    };
+    setErrorStatus(getCurrentErrorStatus);
+  };
+
+  useEffect(() => {
+    setNewInfo(currentInfo);
+  }, [currentInfo]);
+
+  // Field Validation
+  useEffect(() => {
+    let hasError = false;
+    let hasChanges = false;
+    for (const field in currentInfo) {
+      if (currentInfo[field] !== newInfo[field]) {
+        hasChanges = true;
+      }
+      if (
+        (fields.find((n) => n.name === field)?.required && !newInfo[field]) ||
+        errorCases(field, newInfo[field])
+      ) {
+        handleSetError(field, true);
+        if (!hasError) {
+          hasError = true;
+        }
+      } else {
+        handleSetError(field, false);
+      }
+    }
+    setDisableUpdateButton(hasError || !hasChanges);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [newInfo, currentInfo]);
+
+  const handleChange = (event, src) => {
+    const getNewInfo = (currentValue) => {
+      // datetime pickers return value rather than event,
+      // so we can also manually specify target source and value
+      if (src) {
+        let newValue = event;
+        return {
+          ...currentValue,
+          [src]: newValue,
+        };
+      }
+      return {
+        ...currentValue,
+        [event.target.name]:
+          event.target.type === 'checkbox' ? event.target.checked : event.target.value,
+      };
+    };
+
+    let tempNewInfo = getNewInfo(newInfo);
+    if (newInfoCallback) {
+      newInfoCallback(tempNewInfo);
+    }
+    setNewInfo(tempNewInfo);
+  };
+
+  const getFieldLabel = (fieldName) => {
+    const matchingField = allFields.find((field) => field.name === fieldName);
+    return matchingField.label;
+  };
+
+  function getNewFields(currentInfo, newInfo) {
+    const updatedFields = [];
+    Object.entries(newInfo).forEach(([key, value]) => {
+      if (currentInfo[key] !== value)
+        updatedFields.push({
+          Field: key,
+          Value: value,
+          Label: getFieldLabel(key),
+        });
+    });
+    return updatedFields;
+  }
+
+  const handleWindowClose = () => {
+    setOpenConfirmWindow(false);
+    setOpenForm(false);
+    setNewInfo(currentInfo);
+  };
+
+  /**
+   * @param {Array<{name: string, label: string, type: string, menuItems: string[]}>} fields array of objects defining the properties of the input field
+   * @returns JSX correct input for each field based on type
+   */
+  const mapFieldsToInputs = (fields) => {
+    return fields.map((field) => (
+      <InformationField
+        key={field.name}
+        error={field.error}
+        label={field.label}
+        name={field.name}
+        helperText={field.helperText}
+        value={newInfo[field.name]}
+        type={field.type}
+        menuItems={field.menuItems}
```

Comment:
Given that we can condense this function so much, and that it's only ever called within `DialogContent`, and that it's just a wrapper around `fields.map` without actually adding any behavior, I think it would be cleaner to move the map to inline inside of `DialogContent` and get rid of this function entirely.